windows/win_unzip.ps1

```diff
@@ -26,60 +26,40 @@ $result = New-Object psobject @{
     changed = $false
 }
 
-If ($params.creates) {
+$creates = Get-AnsibleParam -obj $params -name "creates"
+If ($creates -ne $null) {
     If (Test-Path $params.creates) {
         Exit-Json $result "The 'creates' file or directory already exists."
     }
-
 }
 
-If ($params.src) {
-    $src = $params.src.toString()
-
-    If (-Not (Test-Path -path $src)){
-        Fail-Json $result "src file: $src does not exist."
-    }
-
-    $ext = [System.IO.Path]::GetExtension($src)
-}
-Else {
-    Fail-Json $result "missing required argument: src"
+$src = Get-AnsibleParam -obj $params -name "src" -failifempty $true
+If (-Not (Test-Path -path $src)){
+    Fail-Json $result "src file: $src does not exist."
 }
 
-If (-Not($params.dest -eq $null)) {
-    $dest = $params.dest.toString()
+$ext = [System.IO.Path]::GetExtension($src)
 
-    If (-Not (Test-Path $dest -PathType Container)){
-        Try{
-            New-Item -itemtype directory -path $dest
-        }
-        Catch {
-            Fail-Json $result "Error creating $dest directory"
-        }
-    }
-}
-Else {
-    Fail-Json $result "missing required argument: dest"
-}
 
-If ($params.recurse) {
-   $recurse = ConvertTo-Bool ($params.recurse)
-}
-Else {
-    $recurse = $false
+$dest = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+If (-Not (Test-Path $dest -PathType Container)){
+    Try{
+        New-Item -itemtype directory -path $dest
+    }
+    Catch {
+        Fail-Json $result "Error creating $dest directory"
```

Comment:
Might be a good thing to return whatever the exception message was here as well as just the 'error creating $dest directory' message here, just to save the user having to debug whether it was because a file already exists, or out of disk space or some other reason.